### PR TITLE
Fix (?) to_pixel in SkyCircle

### PR DESCRIPTION
**Pull request by joleroi**
@keflavich : Can you have a look at this if you have a minute, please?

### CODE REVIEW

File: regions/shapes/circle.py [hunk 1]
```diff
@@ -113,21 +113,8 @@ def to_pixel(self, mywcs, mode='local', tolerance=None):
         if tolerance is not None:
             raise NotImplementedError()
 
-        wcsframe = wcs.utils._wcs_to_celestial_frame_builtin(mywcs)
-        center_in_wcsframe = self.center.transform_to(wcsframe)
-
-        xpix, ypix = mywcs.wcs_world2pix(center_in_wcsframe.spherical.lon,
-                                         center_in_wcsframe.spherical.lat, 0)
```
Comment (joleroi) [inline regions/shapes/circle.py]: These 3 lines are used nowhere

File: regions/shapes/circle.py [hunk 2]
```diff
@@ -113,21 +113,8 @@ def to_pixel(self, mywcs, mode='local', tolerance=None):
...
-
-        if self.radius.unit.physical_type == 'angle':
-            central_pos = coordinates.SkyCoord([mywcs.celestialwcs.crval],
```
Comment (joleroi) [inline regions/shapes/circle.py]: There is clearly a typo here, but I can't figure out what the author wanted to do with this line

File: regions/shapes/circle.py [hunk 3]
```diff
@@ -113,21 +113,8 @@ def to_pixel(self, mywcs, mode='local', tolerance=None):
...
-                                               frame=self.center.name,
-                                               unit=wcs.wcs.cunit)
-            xc, yc, scale, angle = skycoord_to_pixel_scale_angle(central_pos,
-                                                                 wcs)
-            radius_pix = (scale * self.radius).to(u.pixel).value
-        else:  # pixel: this should not be possible.
-            radius_pix = self.radius.value
+        xc, yc, scale, angle = skycoord_to_pixel_scale_angle(self.center, mywcs)
```
Comment (joleroi) [inline regions/shapes/circle.py]: My naive solution

**Comment by joleroi** [comment]
@astrofrog 
any thoughts? :)

**Comment by joleroi** [comment]
Since this is a show stopper, I'll merge now and leave a comment in https://github.com/astropy/regions/issues/13

File: regions/shapes/circle.py (hunk 3 above)
Comment (astrofrog) [inline regions/shapes/circle.py]: Ah yes, maybe ``skycoord_to_pixel_scale_angle`` just deals with any coordinate transformation, in which case this should be fine.

**Comment by astrofrog** [comment]
I think it should be fine - of course, the best test is that if it works, it should be good to go! :)

**Comment by joleroi** [comment]
Define "works" :) I will make a PR to see that a round trip works, and fix the details later, ok?

**Pull request merged**